lib/Compiler.cpp

```diff
@@ -852,63 +875,122 @@ int Compile(const llvm::StringRef &input_filename, const std::string &program,
     return -1;
   }
 
-  // Write the resulting binary.
   // Wait until now to try writing the file so that we only write it on
   // successful compilation.
-  if (output_binary) {
-    output_binary->resize(binary.size() / 4);
-    memcpy(output_binary->data(), binary.data(), binary.size());
+  return WriteOutput(OutputFilename, binaryStream.str().str(), output_buffer);
+}
+
+int CompilePrograms(const std::vector<std::string> &programs,
+                    std::vector<uint32_t> *output_buffer,
+                    std::string *output_log) {
+  std::vector<std::unique_ptr<llvm::Module>> modules;
+
+  llvm::LLVMContext context;
+  for (auto program : programs) {
+    int error;
+    modules.push_back(
+        ProgramToModule(context, "source", program, output_log, &error));
+    if (error != 0)
+      return error;
+  }
+  if (modules.size() == 0 || modules.back() == nullptr)
+    return -1;
```

Comment:
`modules.size() == 0` should only be possible when `programs` is empty, which is likely a misuse of the function so maybe that should be an assert.
If any item in `modules` is `nullptr` and the error value wasn't set to a non-zero value that is also a bug so I think that should be an assert too.